test/parallel/test-net-socket-fd-protection.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+var common = require('../common');
+var net = require('net');
+
+[ 3, 4, 5, 6, 7, 8, 9 ].forEach(function(fd) {
+  try {
+    var stream = new net.Socket({
+      fd: fd,
+      readable: false,
+      writable: true
+    });
+    stream.on('error', function() {});
+    stream.write('might crash');
+  } catch(e) {}
+}); // Should not crash.
```

Comment:
This test isn't durable enough; see https://github.com/nodejs/node/issues/3604#issuecomment-152540585